tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp

```diff
@@ -47,22 +47,24 @@ uint32_t get_read_ptr(bool db_buf_switch) {
     return *reinterpret_cast<volatile tt_l1_ptr uint32_t*>(get_db_cb_rd_ptr_addr(db_buf_switch)) << 4;
 }
 
-inline uint32_t min(uint32_t a, uint32_t b) { return (a < b) ? a : b; }
+//inline uint32_t min(uint32_t a, uint32_t b) { return (a < b) ? a : b; }
 
 FORCE_INLINE void write_buffers(
     volatile tt_l1_ptr uint32_t* command_ptr,
     uint32_t num_destinations,
+    uint32_t num_cores,
     uint32_t consumer_cb_size,
     uint32_t consumer_cb_num_pages,
     uint64_t producer_noc_encoding,
     uint32_t producer_consumer_transfer_num_pages,
     bool db_buf_switch) {
+
+    bool sharded = num_cores > 1;
     for (uint32_t i = 0; i < num_destinations; i++) {
         const uint32_t bank_base_address = command_ptr[1];
         const uint32_t num_pages = command_ptr[2];
         const uint32_t page_size = command_ptr[3];
         const uint32_t dst_buf_type = command_ptr[5];
-        Buffer buffer((BufferType)dst_buf_type, bank_base_address, page_size);
```

Comment:
Do we really need to construct a buffer in the inner loop now? that sounds expensive. it would be ideal if you could fuse the sharded and normal buffer classes, OR create a templated function of the inner loop and case switch with your correct buffer type.